core/src/main/kotlin/org/evomaster/core/problem/graphql/IntrospectiveQuery.kt

```diff
@@ -61,35 +63,38 @@ class IntrospectiveQuery {
                         "query": "fragment FullType on __Type {  kind  name  fields(includeDeprecated: true) {    name    args {      ...InputValue    }   type {      ...TypeRef    }    isDeprecated    deprecationReason  }  inputFields {    ...InputValue  }  interfaces {    ...TypeRef  }  enumValues(includeDeprecated: true) {    name    isDeprecated    deprecationReason  }  possibleTypes {    ...TypeRef  }} fragment InputValue on __InputValue {  name type {   ...TypeRef  }  defaultValue}fragment TypeRef on __Type {  kind  name  ofType {    kind    name   ofType {      kind      name      ofType {        kind       name       ofType {          kind          name         ofType {            kind            name            ofType {             kind             name              ofType {                kind                name              }            }         }        }      }    }  }}query IntrospectionQuery {  __schema {    queryType {     name   }    mutationType {      name    }    types {      ...FullType   }    directives {      name      locations      args {        ...InputValue     }    } }}","variables":null,"operationName":"IntrospectionQuery"
                     }
                 """.trimIndent(), MediaType.APPLICATION_JSON_TYPE)
-
-        //TODO check if TCP problems
-        val response = SearchTimeController.measureTimeMillis({ ms, res ->
+        val response: Response?
+        try {
+            response = SearchTimeController.measureTimeMillis({ ms, res ->
                 LoggingUtil.getInfoLogger().info("Fetched GraphQL schema in ${ms}ms")
-        }, {
-            try {
-                var request = client.target(graphQlEndpoint)
-                    .request("application/json")
+            }, {
+                try {
+                    var request = client.target(graphQlEndpoint)
+                            .request("application/json")
 
-                for (h in list) {
-                    request = request.header(h.first, h.second)
+                    for (h in list) {
+                        request = request.header(h.first, h.second)
+                    }
+                    request.buildPost(query)
+                            .invoke()
+                } catch (e: Exception) {
+                    log.error("Failed query to '$graphQlEndpoint' :  $query")
+                    throw e
```

Comment:
the `throw SutProblemException(` should rather be done here, with `e.message` added to the message of the error, something like:
```
throw SutProblemException("Failed introspection query to '$graphQlEndpoint', please check connection and URL format. Error: ${e.message}" )
```